src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -40,6 +43,21 @@ internal struct ReadStackFrame
         public bool IsProcessingEnumerable => IsEnumerable || IsPropertyEnumerable;
         public bool IsPropertyEnumerable => JsonPropertyInfo != null ? JsonPropertyInfo.ClassType == ClassType.Enumerable : false;
 
+        public bool IsProcessingProperty
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                if (JsonPropertyInfo == null || Skip())
```

Comment:
When is `Drain` ever true? Can you add a comment explaining this `Skip()` (and/or rename it to be self-documenting). Also, maybe make it a property. On initial read, I thought it was doing some work.

Reply:
I just realized `Drain` is another public field (and it set in `HandleArray`/`Object`). We may want to rename both Drain and Skip to be more verbose.

Reply:
I'm not sure why they are separate. @steveharter? Should `Drain` be set to true when we're setting `s_missingProperty`?

Reply:
Skip is intended to be checked. There could be several reasons why a property may be skipped and draining (missing c# property) is one of them.

Reply:
Skip could be made a property. The reason why most are fields is for performance.

Reply:
> There could be several reasons why a property may be skipped and draining (missing c# property) is one of them.

Ok, but it doesn't look like we've added them yet? If `Drain` should be set when we set `s_missingProperty` (which seems right to me?), `Skip()` would be faster as it wouldn't have the second equality check.

Reply:
I believe at the time I was going to hook up the Ignore attribute.

If you can do it all with one bool field then go ahead.

Reply:
> There could be several reasons why a property may be skipped and draining

What are some other reasons?